### PR TITLE
:ghost: add hack to generate jwt token.

### PR DESCRIPTION
**Pull request by jortel**
Add hack script to generate jwt tokens for _builtin_ auth.  Likley used by actors such as the operator for migration related to upgrade.

Related:
- auth.Validator signature changed to return error.  This is more flexible and _failures_ can be propagated.
- auth.Builtin.Authenticate() returns NotAuthenticated when should have returned NotValid for clearity.
- auth.NotValid includes `Reason`.

### CODE REVIEW

**Review by mansam — APPROVED** [review]
nice! works great!

```sh
> ./jwt.sh foobar
eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJzY29wZSI6Iio6KiIsInVzZXIiOiJvcGVyYXRvciJ9.b1oNNnRiqPDD7vQ7ySABnZhb88TTHxQShhUDSY7CEG-DrWg4cy738BU5FsdEuJVnvlOZ3o-VZ5QYqs-PXe5_VQ
```
image

**Review by aufi — APPROVED** [review]
Looks great!

File: auth/builtin.go [hunk 1]
```diff
@@ -71,9 +74,15 @@ type Builtin struct {
 //
 // Authenticate the token
 func (r *Builtin) Authenticate(request *Request) (jwToken *jwt.Token, err error) {
-	token := request.Token
+	token := strings.Replace(request.Token, "Bearer", "", 1)
+	token = strings.Fields(token)[0]
```
Comment (jortel) [inline auth/builtin.go:78]: **hint**: support proper header value:  `Authorization: <type> <token>`.

File: auth/builtin.go [hunk 2]
```diff
@@ -93,32 +102,52 @@ func (r *Builtin) Authenticate(request *Request) (jwToken *jwt.Token, err error)
 	}
 	claims, cast := jwToken.Claims.(jwt.MapClaims)
 	if !cast {
-		err = liberr.Wrap(&NotAuthenticated{Token: token})
+		err = liberr.Wrap(
+			&NotValid{
```
Comment (jortel) [inline auth/builtin.go:106]: **hint**: This is a token validation error, not authentication.

File: auth/provider.go [hunk 3]
```diff
@@ -51,26 +52,29 @@ type NotAuthenticated struct {
...
 
 func (e *NotAuthenticated) Error() (s string) {
-	return fmt.Sprintf("Token %s not-valid.", e.Token)
+	return fmt.Sprintf("Token [%s] not-authenticated.", e.Token)
 }
 
 func (e *NotAuthenticated) Is(err error) (matched bool) {
-	_, matched = err.(*NotAuthenticated)
+	notAuth := &NotAuthenticated{}
+	matched = errors.As(err, &notAuth)
```
Comment (jortel) [inline auth/provider.go:60]: Using `As()` is more robust in case of wrapped errors.

File: task/auth.go [hunk 4]
```diff
@@ -22,26 +24,34 @@ type Validator struct {
 //  - The token references a task.
 //  - The task is valid and running.
 //  - The task pod valid and pending|running.
-func (r *Validator) Valid(token *jwt.Token, db *gorm.DB) (valid bool) {
-	var err error
+func (r *Validator) Valid(token *jwt.Token, db *gorm.DB) (err error) {
```
Comment (mansam) [inline task/auth.go:27]: Really like this refactor.

**Review by mansam — APPROVED** [review]
LGTM

**Pull request merged**